server/scriptController.js

```diff
@@ -19,11 +21,110 @@ const scriptController = {
       })
       .catch((error) => {
         return next({
-          log: `error: ${error} occured when getting script titles`,
-          message: 'error in getScriptTitles in scriptController' ,
+          log: `error: ${error} occurred when getting script titles`,
+          message: 'error in getScriptTitles in scriptController',
         });
       });
   },
+  saveScript: (req, res, next) => {
+    const MAX_FILESIZE_BYTES = 50 * 1024 * 1024; //50MB. If updating, change constant in Upload.jsx too.
+
+    // configure multer storage to upload files to the path from the .env file
+    const storage = multer.diskStorage({
+      destination: function (req, file, cb) {
+        cb(null, process.env.UPLOADPATH);
+      },
+      filename: function (req, file, cb) {
+        cb(null, Date.now() + file.originalname);
+      },
+    });
+
+    function fileFilter(req, file, cb) {
+      const name = file.originalname;
+      const regex = /^[a-zA-Z0-9](?:[a-zA-Z0-9 ._-]*[a-zA-Z0-9])?\.[a-zA-Z0-9_-]+$/;
+      // limit filename size to less than 100 characters
+      // check for invalid file names
+      if (name.length < 100 && name.match(regex)) {
+        return cb(null, true);
+      } else {
+        return cb(new Error('File is incorrect format and was not uploaded'));
```

Comment:
Error should be about the filename issues, not the file itself since only filename stuff is checked here.